### PR TITLE
feat: vitest unit testing base

### PR DESCRIPTION
**Pull request by kevin-hashimoto**
- feat: vitest coverage
- feat: vitest ui
- feat: react testing library setup
- feat: msw setup
- feat: multiple test files with examples that should cover the majority of our tests
- chore: package update

### CODE REVIEW

File: frontend/src/components/Logout.jsx [hunk 1]
```diff
@@ -3,13 +3,13 @@ import BCButton from '@/components/BCButton'
 import { logout } from '@/utils/keycloak'
 import { useCurrentUser } from '@/hooks/useCurrentUser'
 
-const Logout = () => {
+export const Logout = () => {
```
Comment (AlexZorkin) [inline frontend/src/components/Logout.jsx:6]: Is removing the default just a stylistic choice?

File: frontend/src/tests/utils/wrapper.jsx [hunk 2]
```diff
@@ -4,17 +4,21 @@ import { CssBaseline, ThemeProvider } from '@mui/material'
...
+    mutations: {
       retry: false
     }
-  },
-  logger: {
-    log: console.log,
-    warn: console.warn,
-    error: () => {}
   }
+  // logger: {
```
Comment (AlexZorkin) [inline frontend/src/tests/utils/wrapper.jsx:17]: was this too loud? should we just remove it?

File: frontend/src/views/Organizations/AddEditOrg/AddEditOrg.test.jsx
Comment (AlexZorkin) [inline frontend/src/views/Organizations/AddEditOrg/AddEditOrg.test.jsx:1]: are these all left here as reference?

File: frontend/src/views/Organizations/Organizations.test.jsx
Comment (AlexZorkin) [inline frontend/src/views/Organizations/Organizations.test.jsx:1]: same, left for reference?

**Review by AlexZorkin — CHANGES_REQUESTED** [review]
It's nice to get back to baseline! We can leave the commented out tests for later reference. One thing that would be helpful is if you could create a shortlist of the components you think we should add tests for. I will create followup tickets to add coverage on our gaps.

File: frontend/src/components/Logout.jsx (hunk 1 above)
Comment (kevin-hashimoto) [reply frontend/src/components/Logout.jsx:6]: moves it inline with our other components that follow explicit export/imports

File: frontend/src/tests/utils/wrapper.jsx (hunk 2 above)
Comment (kevin-hashimoto) [reply frontend/src/tests/utils/wrapper.jsx]: new react-query version deprecates custom loggers. I've removed the comment

**Comment by kevin-hashimoto** [comment]
> It's nice to get back to baseline! We can leave the commented out tests for later reference. One thing that would be helpful is if you could create a shortlist of the components you think we should add tests for. I will create followup tickets to add coverage on our gaps.

@AlexZorkin added todo tests which should now show up in the test ui as 'skipped'

**Review by AlexZorkin — APPROVED** [review]
Lots todo haha. Thanks for getting us back to baseline.

**Pull request merged**